compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -814,7 +815,7 @@ private void addAssociatedTypeDefinition(BLangConstant constant, BIntersectionTy
 
     private BType createRecordType(BLangExpression expr, BConstantSymbol constantSymbol, Object value, Location pos,
                                    SymbolEnv env) {
-        if (expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+        if (expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF || expr.getKind() == NodeKind.UNARY_EXPR) {
```

Comment:
This seems not to be the root cause for the unary case.

For `{increment: 1}` case we come to `createRecordType()` only once. However, for `{increment: +1}` case we come here twice and the crashing takes place at the second time. 

The reason we come here twice is that, at `evaluateUnaryOperator()` here 
https://github.com/ballerina-platform/ballerina-lang/blob/4339d5dea5f45efc2f4c2cd64c24758e7c431349/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java#L353  

we set the value type pair as follows. 
- value - 1
- type - currentConstSymbol.type = record { increment = 1 }

The type of 1 should be `int`. not the record type. 

That's the same reason `{increment: -1}` case works without this fix and we only crash for +1. 

I think we need to change the type to `value.type` here.

Reply:
Updated as suggested.